File: 242.java
// LeetCode 242 Valid Anagram
// Reference: https://leetcode.com/problems/valid-anagram/?envType=featured-list&envId=top-interview-questions?envType=featured-list&envId=top-interview-questions
import java.util.Arrays;
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {

        // Sorting
        char[] sChar = s.toCharArray();
        char[] tChar = t.toCharArray();

        Arrays.sort(sChar);
        Arrays.sort(tChar);

        return Arrays.equals(sChar, tChar);
    }

    public boolean isAnagram2(String s, String t){

        // HashMap
        HashMap<Character, Integer> count = new HashMap<>();

        // Count the frequency of characters in String s
        for (char x: s.toCharArray()){
            count.put(x, count.getOrDefault(x, 0) + 1);
        }

        // Decrement the frequency of characters in String t
        for (char x: t.toCharArray()){
            count.put(x, count.getOrDefault(x, 0) - 1);
        }

        // Check if any character has non-zero frequency
        for (int val: count.values()){
            if (val != 0){
                return false;
            }
        }

        return true;


    }


}

File: 242.js
// LeetCode 242 Valid Anagram
// Reference:https://leetcode.com/problems/valid-anagram/solutions/3261552/easy-solutions-in-java-python-javascript-and-c-look-at-once/?envType=study-plan-v2&envId=top-interview-150

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) {
        return false;
    }

    var freq = new Array(26).fill(0);

    for (var i = 0; i < s.length; i++){
        freq[s.charCodeAt(i) - 'a'.charCodeAt(0)]++;
        freq[t.charCodeAt(i) - 'a'.charCodeAt(0)]--;
    }

    for (var i = 0; i < freq.length; i++){
        if (freq[i] !== 0){
            return false;
        }
    }
    return true;
};